core/trino-main/src/main/java/io/trino/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -121,6 +124,8 @@ public synchronized void stop()
             future.cancel(true);
             future = null;
         }
+        remoteTaskCleaner.markTaskStatusFetcherStopped(taskStatus.get().getState());
+        dynamicFiltersFetcher.noMoreUpdateRequests();
```

Comment:
could it be just `DynamicFiltersFetcher#stop` method?

Reply:
no - as we want to wait for results of last after last `DFF.updateDynamicFiltersVersionAndFetchIfNecessary` call